### PR TITLE
Add perf numbers for perf_train_cpu

### PR DESCRIPTION
**Pull request by jianan-gu**
# What does this PR do?

As mentioned in https://github.com/huggingface/transformers/pull/17138, we are adding some perf numbers in the doc.

cc @sywangyi  @liangan1

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by jianan-gu** [comment]
@sgugger Could you please review this PR? Thanks!

File: docs/source/en/perf_train_cpu.mdx [hunk 1]
```diff
@@ -58,3 +58,40 @@ Take an example of the use cases on [Transformers question-answering](https://gi
 <b>--use_ipex \</b>
 <b>--bf16 --no_cuda</b></pre> 
 
+### Performance Examples
+
+BERT-base, DistilBERT-base and BERT-large are selected as examples with different model sizes to demonstrate the BFloat16 training performance achieved by IPEX and speed-up ratio against Stock PyTorch FP32 on the 3rd Generation Xeon® Scalable Processors (Intel® Xeon® Platinum 8380H CPU). All the benchmarkings are using [IPEX CPU launcher](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/tutorials/performance_tuning/launch_script.html) to set system configs to ensure a best-to-best comparison, like numactl (--node_id 0 , i.e., one socket full cores) and jemalloc (--enable_jemalloc).
```
Comment (sgugger) [inline docs/source/en/perf_train_cpu.mdx:63]: Like the checkpoint names to their model Hub page on hf.co

File: docs/source/en/perf_train_cpu.mdx [hunk 2]
```diff
@@ -58,3 +58,40 @@ Take an example of the use cases on [Transformers question-answering](https://gi
...
+|-------|------------------------------|
+| PyTorch  | v1.12.1+cpu  |
+| IPEX  | v1.12.100+cpu  |
+|Transformers| [main](https://github.com/huggingface/transformers/commit/0ea53822f8bdc2c0c41b45bcd04fa6c031e5e700) |
+
+
+[Bert-base-cased](https://huggingface.co/bert-base-cased?text=Paris+is+the+%5BMASK%5D+of+France.)
+| Task  | IPEX BFloat16  vs PT Float32 |
+|-------|------------------------------|
+| SST-2 | 1.72x                        |
```
Comment (sgugger) [inline docs/source/en/perf_train_cpu.mdx:82]: What is sped up exactly? Could you maybe just show some concrete metric and have a column IPEX and a column native PyTorch?

**Review by sgugger — COMMENTED** [review]
Thanks for working on this. I think we can make the results a bit clearer.

File: docs/source/en/perf_train_cpu.mdx (hunk 1 above)
Comment (jianan-gu) [reply docs/source/en/perf_train_cpu.mdx]: Have refined the name with the checkpoint names and their link to the model Hub page on hf.co

File: docs/source/en/perf_train_cpu.mdx (hunk 2 above)
Comment (jianan-gu) [reply docs/source/en/perf_train_cpu.mdx]: Have refined the decription of the tables, adding the metric names for a better reading.

File: docs/source/en/perf_train_cpu.mdx [hunk 3]
```diff
@@ -60,38 +60,108 @@ Take an example of the use cases on [Transformers question-answering](https://gi
...
+[Bert-base-cased](https://huggingface.co/bert-base-cased?text=Paris+is+the+%5BMASK%5D+of+France.), [Distilbert-base-uncased](https://huggingface.co/distilbert-base-uncased?text=The+goal+of+life+is+%5BMASK%5D.) and [Bert-large-uncased](https://huggingface.co/bert-large-uncased?text=Paris+is+the+%5BMASK%5D+of+France.) are selected as examples with different model sizes to demonstrate the BFloat16 training performance achieved by IPEX and speed-up ratio against Stock PyTorch FP32 on the 3rd Generation Xeon® Scalable Processors (Intel® Xeon® Platinum 8380H CPU). All the benchmarkings are using [IPEX CPU launcher](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/tutorials/performance_tuning/launch_script.html) to set system configs to ensure a best-to-best comparison, like numactl (--node_id 0 , i.e., one socket full cores) and jemalloc (--enable_jemalloc).
```
Comment (sgugger) [inline docs/source/en/perf_train_cpu.mdx:63]:
```suggestion
[bert-base-cased](https://huggingface.co/bert-base-cased), [distilbert-base-uncased](https://huggingface.co/distilbert-base-uncased) and [bert-large-uncased](https://huggingface.co/bert-large-uncased) are selected as examples with different model sizes to demonstrate the BFloat16 training performance achieved by IPEX and speed-up ratio against Stock PyTorch FP32 on the 3rd Generation Xeon® Scalable Processors (Intel® Xeon® Platinum 8380H CPU). All the benchmarkings are using [IPEX CPU launcher](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/tutorials/performance_tuning/launch_script.html) to set system configs to ensure a best-to-best comparison, like numactl (--node_id 0 , i.e., one socket full cores) and jemalloc (--enable_jemalloc).
```

File: docs/source/en/perf_train_cpu.mdx [hunk 4]
```diff
@@ -60,38 +60,108 @@ Take an example of the use cases on [Transformers question-answering](https://gi
...
-
-[Bert-base-cased](https://huggingface.co/bert-base-cased?text=Paris+is+the+%5BMASK%5D+of+France.)
-| Task  | IPEX BFloat16  vs PT Float32 |
+#### Training configuration:
+| GLUE Task  | Setups |
 |-------|------------------------------|
-| SST-2 | 1.72x                        |
-| STS-B | 1.69x                        |
-| RTE   | 1.68x                        |
+| SST-2; STS-B; RTE |per_device_train_batch_size=128 ; max_seq_len=128 ; num_train_epochs=3 ; learning_rate=2e-5|
```
Comment (sgugger) [inline docs/source/en/perf_train_cpu.mdx:78]: This should be two sentences now, no need for a table here.

File: docs/source/en/perf_train_cpu.mdx [hunk 5]
```diff
@@ -58,3 +58,103 @@ Take an example of the use cases on [Transformers question-answering](https://gi
...
+  <tr>
+    <th rowspan="2">Workload</th>
+    <th colspan="2">Training samples per second</th>
+  </tr>
+  <tr>
+    <th>Task</th>
+    <th>IPEX BFloat16  vs PT Float32</th>
+  </tr>
+  <tr>
+    <td style="text-align: center; vertical-align: middle" rowspan="10">Bert-base-cased</td>
```
Comment (sgugger) [inline docs/source/en/perf_train_cpu.mdx:93]: The names of the model should not be capitalized.

File: docs/source/en/perf_train_cpu.mdx [hunk 6]
```diff
@@ -58,3 +58,103 @@ Take an example of the use cases on [Transformers question-answering](https://gi
...
+  <tr>
+    <th>Task</th>
+    <th>IPEX BFloat16  vs PT Float32</th>
+  </tr>
+  <tr>
+    <td style="text-align: center; vertical-align: middle" rowspan="10">Bert-base-cased</td>    
+  </tr>
+  <tr>
+    <td style="text-align: center; vertical-align: middle">SST-2</td>
+    <td style="text-align: center; vertical-align: middle">1.72x</td>
```
Comment (sgugger) [inline docs/source/en/perf_train_cpu.mdx:97]: This is still not a direct comparison of times. Could you display the times instead of a multiplier?

**Review by sgugger — COMMENTED** [review]
Thanks for iterating. The tables look a bit weird (you can see the results [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_20974/en/perf_train_cpu#performance-speedup-ratio)) and as suggested before, we would need to change the speedups to two columns showing the training times for Ipex and native PyTorch.

File: docs/source/en/perf_train_cpu.mdx (hunk 6 above)
Comment (jianan-gu) [reply docs/source/en/perf_train_cpu.mdx]: Thanks for the suggestions, here have refined this section with several changes on the tables/model names.

For the comparison of times, the absolute time number may need an extra internal procedure if we really want to make them public,  and for now, we could only get the speed-up ratio to be public, here have refined the tables to a diagram. to better present the ratio.

**Comment by jianan-gu** [comment]
Hi @sgugger, 
For the absolute time number to be shown as public in the doc, can we directly mention and link this [public blog](https://huggingface.co/blog/intel-sapphire-rapids) for a reference in this perf_train_cpu doc instead of providing ours? (to simplify this work and avoid extra internal review procedures)

This public blog guides users to run through CPU training and provides realistic results, which is a very practical reference.

Thanks.

**Comment by sgugger** [comment]
Yes, we can definitely link to the blog post instead of the picture.

**Comment by jianan-gu** [comment]
> Yes, we can definitely link to the blog post instead of the picture.
Hi, @sgugger 
Have refined this PR to add the link to this blog post as a practice example in the doc.
Thanks!

**Review by sgugger — APPROVED** [review]

**Comment by sgugger** [comment]
Thanks again for your contribution!
